### PR TITLE
Feature/table type reload fix

### PR DESCRIPTION
**Pull request by JoshLipps**
Fixed a few bugs in TypeTable that were impacting TrafficCounter:

1. Lots of linting
2. changed allChecked logic to use a useEffect instead of a useCallback (prevents calling every render) 
3. Fixed checkbox to actually listen for checked to change in its useEffect
4. Changed Row rendering to be on new row list instead of using useCallback

### CODE REVIEW

File: src/hooks/useCopyToClipboard.tsx [hunk 1]
```diff
@@ -1,32 +1,34 @@
-export const useCopyToClipboard = (text : string) => {
-  return copyToClipboard(text)
-};
+import { useCallback } from 'react';
 
-const copyToClipboard = (str : string) : boolean => {
+export const useCopyToClipboard = () => {
+  const copyToClipboard = useCallback((str : string) : boolean => {
```
Comment (JoshLipps) [inline src/hooks/useCopyToClipboard.tsx:4]: Move copyToClipboard into a `useCallback` that is provided by `useCopyToClipboard` hook

File: src/Tables/molecules/TypeTable.tsx [hunk 2]
```diff
@@ -91,4 +98,8 @@ const TypeTable : React.FC<IProps> = ({ columnConfig, selectable, selectCallback
   );
 };
 
-export default TypeTable;
\ No newline at end of file
+export default TypeTable;
+
+function isChecked({_checked=false} : IRowData){
```
Comment (JoshLipps) [inline src/Tables/molecules/TypeTable.tsx:103]: move this out so it doesn't need to be re declared on re render (its Pure)

File: src/Tables/molecules/TypeTable.tsx [hunk 3]
```diff
@@ -49,18 +49,25 @@ interface IProps {
...
+  selectCallback? : (checked:boolean, id?: string | number)=>void
+  toggleAllCallback? : (checked: boolean)=>void
 }
 
-const TypeTable : React.FC<IProps> = ({ columnConfig, selectable, selectCallback, toggleAllCallback, rows, hasStatus = false, hasThumbnail = false, hasTypeIcon = false }) => {
+const TypeTable : React.FC<IProps> = ({
+  columnConfig,
+  selectable,
+  selectCallback = ()=>{},
+  toggleAllCallback = ()=>{},
```
Comment (JoshLipps) [inline src/Tables/molecules/TypeTable.tsx:60]: Added Default functions for optional functions, the removes the need for "is defined" checks when not specified.

File: src/Tables/molecules/TypeTable.tsx [hunk 4]
```diff
@@ -49,18 +49,25 @@ interface IProps {
   hasStatus?: boolean
   hasThumbnail?: boolean
   hasTypeIcon?: boolean
-  selectCallback? : any
-  toggleAllCallback? : any
+  selectCallback? : (checked:boolean, id?: string | number)=>void
+  toggleAllCallback? : (checked: boolean)=>void
```
Comment (JoshLipps) [inline src/Tables/molecules/TypeTable.tsx:53]: Fixed types for these optional Callbacks.

File: src/Form/atoms/Checkbox.tsx [hunk 5]
```diff
@@ -124,20 +124,22 @@ const Checkbox : React.FC<IProps> = ({ indeterminate = false, disabled, checked
 
     setVisualState( state );
 
-  }, [isChecked, setVisualState])
+  }, [checked, isChecked, setVisualState]);
```
Comment (JoshLipps) [inline src/Form/atoms/Checkbox.tsx:127]: This allows checked to update correctly when external setting changes.

**Review by atomworks — APPROVED** [review]
Looks good! Thanks for tidying up some of those loose ends from the original implementation.

**Pull request merged**